### PR TITLE
Remove Utilities/variant.hpp and use C++17 variant

### PR DESCRIPTION
**Pull request by kukkamario**
- Remove also Utilities/variant_visitor.hpp
- Fix variant and variant_visitor usages and #includes

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKGSRender.cpp [hunk 1]
```diff
@@ -517,7 +517,12 @@ VKGSRender::VKGSRender() : GSRender()
 	display_handle_t display = m_frame->handle();
 
 #if !defined(_WIN32) && !defined(__APPLE__)
-	display.match([this](std::pair<Display*, Window> p) { m_display_handle = p.first; XFlush(m_display_handle); }, [](auto _) {});
+	std::visit([this](auto&& p) {
+		using T = std::decay_t<decltype(p)>;
+		if constexpr (std::is_same_v<T, std::pair<Display*, Window>>) {
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKGSRender.cpp]: The coding style dictates { on newlines

File: rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 #include "stdafx_d3d12.h"
 
+#include <variant>
+
 #include "D3D12GSRender.h"
 #include "d3dx12.h"
 #include "../Common/BufferUtils.h"
 #include "D3D12Formats.h"
 #include "../rsx_methods.h"
 #include "restore_new.h"
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/D3D12/D3D12Buffer.cpp]: restore_new was used for variant header

**Comment by kukkamario** [comment]
Fixed style issues and removed `#include "restore_new.h"`

File: rpcs3/Emu/RSX/RSXThread.h [hunk 3]
```diff
@@ -20,7 +21,6 @@
 #include "Capture/rsx_trace.h"
 #include "Capture/rsx_replay.h"
 #include "restore_new.h"
-#include "Utilities/variant.hpp"
 #include "define_new_memleakdetect.h"
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/RSXThread.h]: define_new_memleakdetect too

Comment (elad335) [reply rpcs3/Emu/RSX/RSXThread.h]: need to remove from rsxthread header as well.

**Pull request merged**